packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -944,19 +940,19 @@ export class DeltaManager
         }
     }
 
-    public cancelDelayInfo(id: string) {
-        this.idToDelayMap.delete(id);
-        this.maxThrottlingDelay = Math.max(...this.idToDelayMap.values());
+    public cancelDelayInfo() {
+        if (Date.now() >= this.timeTillThrottling) {
```

Comment:
This is not entirely correct (at least it telling when we have all connections going).  Let's say we have request for ops and connection failed with 429 asking both to come back in 10 seconds. In 10 seconds, maybe ops request would succeed and we will call into here, but connection to delta ordering service takes10 more seconds, or maybe PUSH is down and it takes another 20 minutes to get there.

If you really want to get rid of IDs, I'd switch to counter and count number of on & off calls.

Also please note that everything related to timers is not exact. Timers may fire sooner or later than their actual target time, due to pressure in the system, but also due to coalescing of timers.

Reply:
Used a set and then when the size of set becomes 0 as all have cancelled their delay, then we set it to 0.